builtin/game/item_entity.lua

```diff
@@ -38,31 +38,32 @@ core.register_entity(":__builtin:item", {
 	age = 0,
 
 	set_item = function(self, itemstring)
-		local stack = ItemStack(itemstring or self.itemstring)
+		itemstring = itemstring or self.itemstring
+		if itemstring == "" then
+			-- item not yet known
+			return
+		end
+		local stack = ItemStack(itemstring)
 		self.itemstring = stack:to_string()
 
 		-- Backwards compatibility: old clients use the texture
 		-- to get the type of the item
-		local itemname = stack:get_name()
+		local itemname = stack:is_known() and stack:get_name() or "unknown"
 
 		local max_count = stack:get_stack_max()
 		local count = math.min(stack:get_count(), max_count)
-		local size = 0.2 + 0.1 * (count / max_count)
-
-		if not stack:is_known() then
-			itemname = "unknown"
-		end
+		local size = 0.2 + 0.1 * (count / max_count)^(1/3)
 
 		self.object:set_properties({
 			is_visible = true,
 			visual = "wielditem",
 			textures = {itemname},
 			visual_size = {x = size, y = size},
 			collisionbox = {-size, -size, -size, size, size, size},
-			automatic_rotate = math.pi * 0.5,
+			automatic_rotate = true,
```

Comment:
`automatic_rotate` was documented wrong. It's actually a float value, which specifies the rotation speed per second.

Reply:
l've tested it and it always had the same rotation speed.

Reply:
sorry, l started the wrong minetest, you're right.